juju/juju
# Migrate utils.UUID to uuid.UUID

This is the start of bringing in the uuid code from the utils package
into the internal package. We'll be slowly moving away from utils
package and this is a step in the right direction.

One of the driving reasons for this change is so that we can correctly 
own the code without fear of changing it and breaking other users
of the code. 
This is the code that juju depends on and we should own that without 
the constant churn of keeping our dependencies up to date. We spend 
way too much time doing that and it's becoming a hindrance when 
trying to deliver features.

## Checklist




## QA steps

This was a very simple case of find and replace, the only tricky part
was leases code in the domain package. In that regard, it's wise to
test that we correctly work with the leases.

```sh
$ juju bootstrap lxd test --build-agent
$ juju add-model default
$ juju deploy ubuntu -n 3
```

Ensure that one of the units is the leader.


## Links

**Jira card:** JUJU-5441
